pkg/capacity/manager.go

```diff
@@ -72,7 +86,8 @@ type ClusterManager struct {
 // Let's say that the limit for member1 is 1000 and for member2 is 2000, then the batch of spaces would be 50 for member1 and 100 for member2.
 //
 // If the preferredCluster is provided and it is also one of the available clusters, then the same name is returned.
-func (b *ClusterManager) GetOptimalTargetCluster(preferredCluster, namespace string) (string, error) {
+// In case the preferredCluster was not provided or not found/available and the clusterRoles are provided then the candidates optimal cluster pool will be made out by only those matching the labels, if any available.
+func (b *ClusterManager) GetOptimalTargetCluster(optimalClusterFilter *OptimalTargetClusterFilter) (string, error) {
```

Comment:
it's just a detail, but it doesn't have to be passed as a pointer - the value should be totally fine
```suggestion
func (b *ClusterManager) GetOptimalTargetCluster(optimalClusterFilter OptimalTargetClusterFilter) (string, error) {
```

Reply:
out of curiosity - what is the benefit of introducing the new type `OptimalTargetClusterFilter` instead of taking 3 parameters?
```suggestion
func (b *ClusterManager) GetOptimalTargetCluster(preferredCluster, toolchainStatusNamespace string, clusterRoles []string) (string, error) {
```
I didn't notice that there would be some particular place where the new type really simplifies the code.

Reply:
I have removed the pointer and now the struct is passed by value, see [6b01926](https://github.com/codeready-toolchain/host-operator/pull/736/commits/6b01926137a7f5c783ed0b886e61eaba8cc13642)

The advantages IMO for introducing the struct are mainly those 2:

1. you can avoid setting blank/empty values for optional parameters, for example when there is no `preferredCluster` and no `clusterRoles` the code will look like:
```
manager.GetOptimalTargetCluster("", toolchainStatusNamespace, []string{})
```
which was a bit weird from my point of view.

2. you can add more optional parameters without changing the caller code if not required.

I agree that it's more benefitting for the test code, but those were the main reasons for introducing this new struct.